libgadget/sfr_eff.c

```diff
@@ -677,26 +677,27 @@ static int make_particle_wind(MyIDType ID, int i, double v, double vmean[3]) {
 
 static int make_particle_star(int i) {
     double mass_of_star = find_star_mass(i);
+    int child;
     if(P[i].Type != 0)
         endrun(7772, "Only gas forms stars, what's wrong?");
 
-    /* if we get all mass or a fraction */
-    int child = slots_fork(i, 4);
-
+    /*Store the index of the SPH particle properties as overwritten in slots_convert*/
+    int oldslot = P[i].PI;
```

Comment:
I feel it is better to save the properties rather than a fragile index here. It avoids direct access to the SphP array.